src/invidious/routes/login.cr

```diff
@@ -434,6 +434,13 @@ module Invidious::Routes::Login
 
         sid = Base64.urlsafe_encode(Random::Secure.random_bytes(32))
         user, sid = create_user(sid, email, password)
+
+        if language_header = env.request.headers["Accept-Language"]?
+          if language = ANG.language_negotiator.best(language_header, LOCALES.keys)
+            user.preferences.locale = language.header
+          end
+        end
```

Comment:
Why are you parsing the language header in here?

Reply:
Most of the logic was directly taken from #2299 which parses the language header to update the user preferences here.
https://github.com/iv-org/invidious/blob/3a759bdafbb0077fda62cbb463b232f8675b3c28/src/invidious/routes/login.cr#L438-L445

But you're right, this is unnecessary. The `before_all` section should already update the user preferences